user.go

```diff
@@ -348,124 +348,144 @@ func (u *user) run() {
 		go network.run()
 	}
 
-	for e := range u.events {
-		switch e := e.(type) {
-		case eventUpstreamConnected:
-			uc := e.uc
+	detachTicker := time.NewTicker(10 * time.Second)
 
-			uc.network.conn = uc
+	for {
+		select {
+		case e := <-u.events:
+			switch e := e.(type) {
```

Comment:
Maybe it's a good time to split this `switch` into a separate function…

Reply:
Not needed IMO